common/src/main/java/com/wynntils/wynn/model/PlayerRelationsModel.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.ChatPacketReceivedEvent;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.RelationsUpdateEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import net.minecraft.network.chat.ChatType;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+/**
+ * This model handles the player's in-game relations, like friends, party info, guild info.
+ */
+public class PlayerRelationsModel extends Model {
+    private static final Pattern FRIEND_LIST_MESSAGE_PATTERN = Pattern.compile(".+'s friends \\(.+\\): (.*)");
+    private static final Pattern FRIEND_NO_LIST_MESSAGE_PATTERN_1 = Pattern.compile("§eWe couldn't find any friends.");
+    private static final Pattern FRIEND_NO_LIST_MESSAGE_PATTERN_2 =
+            Pattern.compile("§eTry typing §r§6/friend add Username§r§e!");
+    private static final Pattern FRIEND_REMOVE_MESSAGE_PATTERN =
+            Pattern.compile("§e(.+) has been removed from your friends!");
+    private static final Pattern FRIEND_ADD_MESSAGE_PATTERN = Pattern.compile("§e(.+) has been added to your friends!");
+
+    private static final Pattern PARTY_LIST_MESSAGE_PATTERN = Pattern.compile("Party members: (.*)");
+    private static final Pattern PARTY_NO_LIST_MESSAGE_PATTERN = Pattern.compile("§eYou must be in a party to list.");
+    private static final Pattern PARTY_OTHER_LEAVE_MESSAGE_PATTERN = Pattern.compile("§e(.+) has left the party.");
+    private static final Pattern PARTY_OTHER_JOIN_MESSAGE_PATTERN = Pattern.compile("§e(.+) has joined the party.");
+    private static final Pattern PARTY_SELF_LEAVE_MESSAGE_PATTERN =
+            Pattern.compile("§eYou have been removed from the party.");
+    private static final Pattern PARTY_SELF_JOIN_MESSAGE_PATTERN =
+            Pattern.compile("§eYou have successfully joined the party.");
+    private static final Pattern PARTY_DISBAND = Pattern.compile("§eYour party has been disbanded.");
+
+    private static boolean expectingFriendMessage = false;
```

Comment:
Do you think we should have a timeout for this? If something goes wrong, we'll be looking for these replies for ever. And then the user types `/friends` manually, and the result is eaten up here.

I'm not sure about this. A timeout complicates code, and it can instead cause the problem "what if the reply is just a bit later than the timeout?". So this is mostly me thinking loud. I'll leave the decision to do this, or not, to you.

Reply:
Wynn can lag quite a bit, text messages can be delayed by seconds. I think eating up a /party list or /friend list is not too bad.